### DISCUSSION TITLE
What is the function of the WRITE options when given to Files.newOutputStream

### DISCUSSION THREAD

**Discussion by kuiux** [community/community#116819] [Projects and Issues] [answered]
### Select Topic Area

Question

### Feature Area

Issues

### Body

When opening an OutputStream using the Files API you can pass options for it. One of the options is WRITE which is applied by default if no options are specified. However I fail to see what the difference is when it is provided and when it is not. An outputstream can only be used for writing, no? 

An outputstream is per definition always to write content so I fail to so the use for it. In our project (where we had a bug related to the TRUNCATE_EXISTING missing) the WRITE option was not specified but the file was (over)written either anyway. To be clear, what is the difference between:

```
// OpenOption.CREATE, OpenOption.TRUNCATE_EXISTING, OpenOption.WRITE
Files.newOutputStream(file); 
```
and
`Files.newOutputStream​(file, OpenOption.CREATE, OpenOption.TRUNCATE_EXISTING);`

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by iamsuiux** [comment, marked as answer]
In the context of using the Files API in Java to open an OutputStream, specifying options like WRITE, CREATE, or TRUNCATE_EXISTING can fine-tune how the file is accessed and manipulated.

By default, if no options are provided to Files.newOutputStream, the WRITE option is indeed implied. This means the stream is opened with the basic expectation that you intend to write to the file. The underlying assumption is that you're going to output data to the file, but it doesn't specify how the file should be treated if it already exists or needs to be created.

In your project's case, the bug related to missing TRUNCATE_EXISTING probably meant that when you expected a file to be overwritten from scratch (content fully replaced), it wasn't happening as intended without this option. The file was being written to, but without truncating, leading to unexpected results. Adding TRUNCATE_EXISTING ensures the file starts "clean" with each write operation.
